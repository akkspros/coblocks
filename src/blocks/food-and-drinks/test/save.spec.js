/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { registerBlockType, createBlock, serialize } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;
let serializedBlock;

describe( name, () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );

		// Reset the reused variables.
		serializedBlock = '';
	} );

	it( 'should render', () => {
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render hide images', () => {
		block.attributes.showImages = 0;
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '{\"showImages\":0}' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render hide prices', () => {
		block.attributes.showPrices = 0;
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '{\"showPrices\":0}' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render 3 columns', () => {
		block.attributes.columns = 3;
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'data-columns=\"3\"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );
} );
