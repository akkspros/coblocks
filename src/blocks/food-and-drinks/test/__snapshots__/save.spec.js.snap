// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/food-and-drinks should render 1`] = `
"<!-- wp:coblocks/food-and-drinks -->
<div class=\\"wp-block-coblocks-food-and-drinks\\" data-columns=\\"2\\" itemscope itemtype=\\"http://schema.org/Menu\\"></div>
<!-- /wp:coblocks/food-and-drinks -->"
`;

exports[`coblocks/food-and-drinks should render 3 columns 1`] = `
"<!-- wp:coblocks/food-and-drinks {\\"columns\\":3} -->
<div class=\\"wp-block-coblocks-food-and-drinks\\" data-columns=\\"3\\" itemscope itemtype=\\"http://schema.org/Menu\\"></div>
<!-- /wp:coblocks/food-and-drinks -->"
`;

exports[`coblocks/food-and-drinks should render hide images 1`] = `
"<!-- wp:coblocks/food-and-drinks {\\"showImages\\":0} -->
<div class=\\"wp-block-coblocks-food-and-drinks\\" data-columns=\\"2\\" itemscope itemtype=\\"http://schema.org/Menu\\"></div>
<!-- /wp:coblocks/food-and-drinks -->"
`;

exports[`coblocks/food-and-drinks should render hide prices 1`] = `
"<!-- wp:coblocks/food-and-drinks {\\"showPrices\\":0} -->
<div class=\\"wp-block-coblocks-food-and-drinks\\" data-columns=\\"2\\" itemscope itemtype=\\"http://schema.org/Menu\\"></div>
<!-- /wp:coblocks/food-and-drinks -->"
`;
